barman/utils.py

```diff
@@ -614,6 +614,24 @@ def check_positive(value):
     return int_value
 
 
+def check_tli(value):
+    """
+    Check for a positive integer option, and also make "current" and "latest" acceptable values
+
+    :param value: str containing the value to check
+    """
+    retval = value
+    if value is None:
+        return None
+    try:
+        retval = check_positive(value)
+    except ArgumentTypeError:
+        #  Value is not a positive integer so check for the valid string alternatives here
+        if value not in ["current", "latest"]:
+            raise ArgumentTypeError("'%s' is not a valid TLI" % value)
+    return retval
```

Comment:
This could be further simplified if we check the allowed string values first. That way we don't have to do the `retval = value` assignment and can also avoid using the exception to handle normal behaviour (apologies, I know that was my suggestion from the last review), e.g.:

```
if value in ["current", "latest"]
    return value
try:
    return check_positive(value)
except ArgumentTypeError:
    raise ArgumentTypeError("'%s' is not a valid TLI" % value)
```